src/infra/core/src/pull_service_impl.rs

```diff
@@ -495,20 +517,17 @@ impl PullService for PullServiceImpl {
     ) -> Result<Vec<PullResponse>, InternalError> {
         let requests = dataset_refs
             .into_iter()
-            .map(|r| {
-                // TODO: Support local multi-tenancy
-                match r.as_local_single_tenant_ref() {
-                    Ok(local_ref) => PullRequest {
-                        local_ref: Some(local_ref),
-                        remote_ref: None,
-                        ingest_from: None,
-                    },
-                    Err(remote_ref) => PullRequest {
-                        local_ref: None,
-                        remote_ref: Some(remote_ref),
-                        ingest_from: None,
-                    },
-                }
+            .map(|r| match self.classify_dataset_reference(&r) {
+                Ok(local_ref) => PullRequest {
+                    local_ref: Some(local_ref),
+                    remote_ref: None,
+                    ingest_from: None,
+                },
+                Err(remote_ref) => PullRequest {
```

Comment:
Would it make sense to extract this code into `PullRequest::from_any_ref()` to avoid duplication?

Reply:
Good point, extracted.